graylog2-web-interface/src/views/stores/CurrentViewStateStore.test.js

```diff
@@ -100,7 +100,42 @@ describe('CurrentViewStateStore', () => {
     ViewStatesActions.update = updateFn;
     CurrentViewStateStore.onViewStoreChange({ activeQuery: viewId, view: oldViewState });
     CurrentViewStateStore.onViewStatesStoreChange(sMap);
-    CurrentViewStateStore.widgets(expectedWidgets);
+    await CurrentViewStateStore.widgets(expectedWidgets);
+    expect(updateFn).toHaveBeenCalledTimes(1);
+  });
+
+  it('should remove widget positions for deleted widgets', async () => {
+    const widgetOnePos = new WidgetPosition(1, 1, 5, 6);
+    const widgetTwoPos = new WidgetPosition(1, 6, 5, 6);
+    const widgetPositionsMap = { 'widget-one': widgetOnePos, 'widget-two': widgetTwoPos };
+    const widgetOne = MessagesWidget.builder().id('widget-one').build();
+    const widgetTwo = MessagesWidget.builder().id('widget-two').build();
+    const existingWidgets = [widgetOne, widgetTwo];
+    const existingViewState = viewState.toBuilder()
+      .widgetPositions(widgetPositionsMap)
+      .widgets(existingWidgets)
+      .build();
+
+    viewStateMap[viewId] = existingViewState;
+    const sMap = Immutable.Map(viewStateMap);
+
+    const expectedWidgets = [widgetOne];
+    const expectedWidgetPosition = { 'widget-one': widgetOnePos };
+    const expectedViewState = viewState.toBuilder()
+      .widgetPositions(expectedWidgetPosition)
+      .widgets(expectedWidgets)
+      .build();
+
+    const updateFn = mockAction(jest.fn((id, newViewState) => {
+      expect(id).toEqual(viewId);
+      expect(newViewState).toEqual(expectedViewState);
+      return Promise.resolve(expectedViewState);
+    }));
+
+    ViewStatesActions.update = updateFn;
+    CurrentViewStateStore.onViewStoreChange({ activeQuery: viewId, view: existingViewState });
```

Comment:
Setup of the test case and the actual action(s) which are tested should be clearer, by adding new lines in between. Imo assigning `ViewStatesActions.update` is still part of the setup, I am not sure about this line. I think the actual action under test is `CurrentViewStateStore.widgets`, so it should be visually separated from the setup and checking the expectations afterwards.